### PR TITLE
callautomation private review internal revision

### PR DESCRIPTION
**Pull request by wangrui-msft**
1. Add text source constructor
2. Add polymorphism to recognizecomplete event

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/RecognizeCompleted.cs [hunk 1]
```diff
@@ -52,7 +81,30 @@ public static RecognizeCompleted Deserialize(string content)
             using var document = JsonDocument.Parse(content);
             JsonElement element = document.RootElement;
 
-            return DeserializeRecognizeCompleted(element);
+            RecognizeCompleted parsedRecognizeCompleted = DeserializeRecognizeCompleted(element);
```
Comment (cochi2) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/RecognizeCompleted.cs:84]: why do we need to create the object twice? is not possible to add the below logic within DeserializerRecognizeCompleted?

Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/RecognizeCompleted.cs:84]: yes, the reviewer require the recognizeComplete event only populate one result. Right now, the default recognizecomplete event populate two results as default behavior: 1. dtmf result 2. choice result. However, at the same time, there should only be one result be populated. That's why we add one more processing on this part.

Comment (cochi2) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/RecognizeCompleted.cs:84]: Is the default on an autogenerated class that we cannot modify?

**Review by cochi2 — APPROVED** [review]

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/RecognizeCompleted.cs [hunk 2]
```diff
@@ -32,7 +32,36 @@ internal RecognizeCompleted(string callConnectionId, string serverCallId, string
...
+        public RecognizeCompleted(string callConnectionId, string serverCallId, string correlationId, string operationContext, ResultInformation resultInformation, CallMediaRecognitionType recognitionType, RecognizeResult recognizeResult)
+        {
+            CallConnectionId = callConnectionId;
+            ServerCallId = serverCallId;
+            CorrelationId = correlationId;
+            OperationContext = operationContext;
+            ResultInformation = resultInformation;
+            RecognitionType = recognitionType;
             ReasonCode = new ReasonCode(resultInformation.SubCode.ToString());
+            if (recognizeResult is CollectTonesResult collecToneResult)
```
Comment (Paresh-Arvind-Patil) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/RecognizeCompleted.cs:57]: Nit: seems like a typo collectToneResult ?

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/RecognizeCompleted.cs (hunk 1 above)
Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/RecognizeCompleted.cs:84]: > Is the default on an autogenerated class that we cannot modify?

Yes, correct, the event schema make the RecognizeComplete not allowed to modify on serialize/deserialize

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/RecognizeCompleted.cs (hunk 2 above)
Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/RecognizeCompleted.cs:57]: good catch, will fix it

**Pull request merged**